src/averageAges.js

```diff
@@ -55,7 +67,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+    && people.some(mom => mom.name === person.mother))
```

Comment:
even if you write it this way
```suggestion
    ? people.filter(
    		person => person.sex === 'm'
    		&& people.some(mom => mom.name === person.mother)
    	)
```

Reply:
don't forget about formatting properly

Reply:
plus, make sure you don't combine getting by sex and other query in the same `filter` – make those processes separate, like you did with `men`